stwcs/wcsutil/headerlet.py

```diff
@@ -1344,7 +1351,10 @@ def _delete_single_headerlet(filename, hdrname=None, hdrext=None, distname=None,
     wcscorr.delete_wcscorr_row(fobj['WCSCORR'].data, selections)
 
     # delete the headerlet extension now
-    for hdrind in hdrlet_ind:
+    hdrlet_ind.reverse()
+    del_all = 0 if delete_all else 1
+    del_hdrlets = slice(0, len(hdrlet_ind) - del_all)
+    for hdrind in hdrlet_ind[del_hdrlets]:
```

Comment:
Why not this simple:
```suggestion
    del hdrlet_ind[int(keep_first):]
```
Am I missing something?

Reply:
The multiple lines of logic I implemented work to correct the problem you are trying to re-introduce with your suggestion.  If you have a list `a=['x','y','z','0']` and run the command `del a[1]` you will get `a=['x','z','0']`, but if you want to del both 'y' and 'z' at positions 1 and 2, then run the command `del a[1]; del a[2]`, you will get `a=['x','z']` instead since the indices change upon each deletion.  By reversing the indices, we avoid this problem and delete the desired elements, which is what my code does.  The second suggestion, though, does seem to be more efficient, and I will see about folding that into the next commit after using `.reverse()`.

Reply:
I promise I am not _trying_ to re-introduce the problem. But if it does not work, it does not work. Sorry.

Reply:
I think my mistake was:
```python
del hdrlet_ind[1 if keep_first else 0:]
```
instead it should have been
```python
for hdrind in hdrlet_ind[1 if keep_first else 0:]
```
I still no not think `reverse()` does anything special: dropping last element in a reversed list is the same as starting with the second element in the original list.

Reply:
The use of `.reverse()` is one of the primary changes necessary to get the correct behavior because of how it interacts with lists.  Please review my example using a simple list of 5 elements to see what I am talking about.  This is affects the output when 2 or more extensions need to be deleted, something I have already seen in test cases.